### PR TITLE
TOC, dev/reference, dev/how-to: update several documents

### PR DESCRIPTION
**Pull request by ericsyh**
### What is changed, added or deleted? 

This PR update/add multi documents in :
1. Add string / cluster charset, update some info in **dev/reference/sql/character-set.md**
2. Update the usage of `admin show slow` in **dev/how-to/maintain/identify-slow-queries.md**
3. Add a new doc about sql-mode in **dev/reference/sql-mode.md**, and update the TOC
4. Update some configs in **dev/reference/configuration/tidb-server/configuration.md**
5. Update some configs in **dev/reference/configuration/tidb-server/configuration-file.md**
6. Update some configs in **dev/reference/configuration/pd-server/configuration.md**
7. Add a new doc in **dev/reference/configuration/pd-server/configuration-file.md** and update the TOC
8. Add a new doc in **dev/reference/configuration/tidb-server/configuration-file.md** and update the TOC

### CODE REVIEW

**Comment by ericsyh** [comment]
@lilin90 @scsldb PTAL

File: dev/reference/sql/sql-mode.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## 主要重要 sql_mode 值为：
+
+* ANSI: 符合标准 SQL ，对数据进行校验，如果不符合定义类型或长度，对数据类型调整或截断保存，且返回warning警告。
+* STRICT_TRANS_TABLES:  严格模式，对数据进严格校验，但数据出现错误时，插入到表中，并且返回错误。
+* TRADITIONAL: 采用此模式使 TiDB 的行为象 "传统" SQL 数据库系统，当在列中插入不正确的值时“给出错误而不是警告”，一旦发现错误立即放弃INSERT/UPDATE。
+
+## SQL mode 列表，如下：
+
+| **名称** | **含义** |
```
Comment (lilin90) [inline dev/reference/sql/sql-mode.md]:
```suggestion
| 名称 | 含义 |
```

File: dev/reference/sql/sql-mode.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+TiDB 服务器采用不同 SQL 模式来操作，且不同客户端可以应用不同模式。SQL 模式定义 TiDB 支持哪些 SQL 语法及执行哪种数据验证检查.
+
+TiDB 启动之前采用修改 ` --sql-mode="modes" ` 配项设置 SQL 模式。
+
+TiDB 启动之后采用 `SET [ SESSION | GLOBAL ] sql_mode='modes' `设置 SQL 模式。设置 GLOBAL 级别的 SQL 模式时用户需要有 SUPER 权限，并且只会影响到从设置 SQL 模式开始后续新建立的连接（注：老连接不受影响)。 SESSION 级别的 SQL 模式的变化只会影响当前的客户端。
+
+Modesis是用逗号 (', ') 间隔开的一系列不同的模式。使用 `SELECT @@sql_mode` 语句查询当前 SQL 模式，SQL 模式默认值：""。
+
+## 主要重要 sql_mode 值为：
```
Comment (lilin90) [inline dev/reference/sql/sql-mode.md]: **DO NOT use punctuation in titles except that in FAQ.**
```suggestion
## 重要的 sql_mode 值
```

**Comment by lilin90** [comment]
Since we DO NOT keep a `TOC.md` file at the root directory of this repo, could you update it based on the latest directory? And update the corresponding files in both `dev` and `v3.0`? Now we have multiple versions.

File: dev/reference/configuration/pd-server/configuration-file.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
+---
+title: PD 配置文件描述
+category: reference
+aliases: ['/docs-cn/op-guide/pd-config-file/']
```
[the remark this answers is not in the document]
Comment (ericsyh) [reply dev/reference/configuration/pd-server/configuration-file.md]: Removed.

File: dev/reference/sql/sql-mode.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+title: SQL Mode
+category: reference
+aliases: ['/docs-cn/dev/reference/sql-mode']
```
[the remark this answers is not in the document]
Comment (ericsyh) [reply dev/reference/sql/sql-mode.md]: Removed

File: dev/how-to/maintain/identify-slow-queries.md [hunk 5]
```diff
@@ -254,3 +254,21 @@ admin show slow top all 5
...
 
 由于内存限制，保留的慢查询记录的条数是有限的。当命令查询的 `N` 大于记录条数时，返回的结果记录条数会小于 `N`。
+
+输出内容详细说明，如下:
+   
+| 列名 | 描述 |
+|:------|:---- |
+| start | SQL 语句执行开始时间 |
+| duration | SQL 语句执行持续时间 |
+| details |   |
```
[the remark this answers is not in the document]
Comment (ericsyh) [reply dev/how-to/maintain/identify-slow-queries.md]: Updated  in https://github.com/pingcap/docs-cn/pull/1495/commits/4eaa160651b93765d6b0d1fe8bd0cc3a878e203d

**Comment by ericsyh** [comment]
@lilin90 I apply the changes in dev directory and v3.0 directory.

File: dev/reference/configuration/pd-server/configuration.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 ---
 title: PD 配置参数
 category: reference
+aliases: ['/docs-cn/op-guide/pd-configuration/']
```
Comment (dcalvin) [inline dev/reference/configuration/pd-server/configuration.md]: Alias only in 3.0 folder. 
```suggestion
```

Comment (ericsyh) [reply dev/reference/configuration/pd-server/configuration.md]: okay, let me update.

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 7]
```diff
@@ -182,51 +233,154 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ TiKV 的负载阈值，如果超过此阈值，会收集更多的 batch 封包，来减轻 TiKV 的压力。仅在 `tikv-client.max-batch-size` 值大于 0 时有效，不推荐修改该值。
++ 默认值：200
+
 ## txn-local-latches
 
 事务内存锁相关配置，当本地事务冲突比较多时建议开启。
 
 ### `enable`
 
 + 开启
```
Comment (dcalvin) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 开启或关闭事务内存锁
```

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 8]
```diff
@@ -182,51 +233,154 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
-+ 默认：1024000
++ 默认值：1024000
+
+## binlog
+
+TiDB Binlog 相关配置。
+
+### `enable`
+
++ 开启 binlog 开关。
```
Comment (dcalvin) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ binlog 开关。
```

File: dev/reference/sql/character-set.md [hunk 9]
```diff
@@ -25,6 +25,10 @@ mysql> SHOW CHARACTER SET;
 +---------|---------------|-------------------|--------+
 5 rows in set (0.00 sec)
 ```
+> **注意：**
+> 
+> - 在 `TiDB` 中 `utf8` 被当做成了 `utf8mb4` 来处理。
+> - 每种字符集都对应一个默认的Collation，当前有且仅有一个
```
Comment (dcalvin) [inline dev/reference/sql/character-set.md]:
```suggestion
> - 每种字符集都对应一个默认的 Collation，当前有且仅有一个。
```

File: dev/reference/sql/sql-mode.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
+---
+title: SQL Mode
+category: reference
+
```
Comment (dcalvin) [inline dev/reference/sql/sql-mode.md]:
```suggestion
```

File: v3.0/reference/configuration/pd-server/configuration-file.md [hunk 11]
```diff
@@ -0,0 +1,245 @@
+---
+title: PD 配置文件描述
+category: reference
+
```
Comment (dcalvin) [inline v3.0/reference/configuration/pd-server/configuration-file.md]:
```suggestion
```

File: v3.0/how-to/maintain/identify-slow-queries.md [hunk 12]
```diff
@@ -1,7 +1,6 @@
 ---
 title: 慢查询日志
 category: how-to
-aliases: ['/docs-cn/sql/slow-query/']
```
Comment (dcalvin) [inline v3.0/how-to/maintain/identify-slow-queries.md]: why remove this?

File: v3.0/reference/sql/sql-mode.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
+---
+title: SQL Mode
+category: reference
+
```
Comment (dcalvin) [inline v3.0/reference/sql/sql-mode.md]:
```suggestion
```

File: v3.0/reference/sql/character-set.md [hunk 14]
```diff
@@ -26,6 +25,10 @@ mysql> SHOW CHARACTER SET;
 +---------|---------------|-------------------|--------+
 5 rows in set (0.00 sec)
 ```
+> **注意：**
+> 
+> - 在 `TiDB` 中 `utf8` 被当做成了 `utf8mb4` 来处理。
+> - 每种字符集都对应一个默认的Collation，当前有且仅有一个
```
Comment (dcalvin) [inline v3.0/reference/sql/character-set.md]:
```suggestion
> - 每种字符集都对应一个默认的 Collation，当前有且仅有一个。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 15]
```diff
@@ -0,0 +1,1046 @@
...
+### `block-cache-size`
+
++ rocksdb block cache size。
++ 默认值：机器总内存 / 4
++ 最小值：0
++ 单位：KB|MB|GB
+
+### `disable-block-cache`
+
++ 开启 block cache 开关。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 开启或关闭 block cache。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 16]
```diff
@@ -0,0 +1,1046 @@
...
++ 默认值：false
+
+### `cache-index-and-filter-blocks`
+
++ 开启 缓存 index 和 filter 的开关。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 开启或关闭缓存 index 和 filter。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 17]
```diff
@@ -0,0 +1,1046 @@
...
+### `block-size`
+
++ rocksdb block size。
++ 默认值：64KB
++ 最小值：1KB
++ 单位：KB|MB|GB
+
+### `block-cache-size`
+
++ rocksdb block cache size。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 设置 rocksdb block 缓存大小。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 18]
```diff
@@ -0,0 +1,1046 @@
...
++ 默认值：1
++ 最小值：1
+
+## rocksdb.defaultcf
+
+rocksdb defaultcf 相关的配置项。
+
+### `block-size`
+
++ rocksdb block size。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 设置 rocksdb block 大小。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 19]
```diff
@@ -0,0 +1,1046 @@
...
++ 日志存储目录。
++ 默认值：""
+
+## rocksdb.titan
+
+Titan 相关的配置项。
+
+### `enabled`
+
++ 开启 Titan 开关。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 开启或关闭 Titan。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 20]
```diff
@@ -0,0 +1,1046 @@
...
++ 默认值：false
+
+### `dirname`
+
++ Titan Blob 文件存储目录。
++ 默认值：titandb
+
+### `disable-gc`
+
++ 关闭 Titan 对 Blob 文件的 GC 的开关。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 控制是否关闭 Titan 对 Blob 文件的 GC。
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 21]
```diff
@@ -0,0 +1,1046 @@
...
++ 单位：B|KB|MB|GB
+
+### `enable-statistics`
+
++ 开启自动优化 Rate LImiter 的配置的开关。
++ 默认值：false
+
+### `stats-dump-period`
+
++ 开启 Pipelined Write 的开关。
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 开启或关闭 Pipelined Write。
```

File: v3.0/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 22]
```diff
@@ -1,7 +1,6 @@
 ---
 title: TiDB 专用系统变量和语法
 category: reference
-aliases: ['/docs-cn/sql/tidb-specific/']
```
Comment (dcalvin) [inline v3.0/reference/configuration/tidb-server/tidb-specific-variables.md]: add it back

File: v3.0/reference/configuration/tidb-server/mysql-variables.md [hunk 23]
```diff
@@ -1,7 +1,6 @@
 ---
 title: 系统变量
 category: reference
-aliases: ['/docs-cn/sql/variable/']
```
Comment (dcalvin) [inline v3.0/reference/configuration/tidb-server/mysql-variables.md]: add it back,

File: v3.0/reference/configuration/tidb-server/configuration.md [hunk 24]
```diff
@@ -1,7 +1,6 @@
 ---
 title: TiDB 配置参数
 category: reference
-aliases: ['/docs-cn/op-guide/configuration/','/docs-cn/sql/server-command-option/','/docs-cn/sql/tidb-server/']
```
Comment (dcalvin) [inline v3.0/reference/configuration/tidb-server/configuration.md]: add it back

File: v3.0/how-to/maintain/identify-slow-queries.md (hunk 12 above)
Comment (ericsyh) [reply v3.0/how-to/maintain/identify-slow-queries.md]: Sorry, action by mistake. Add back in https://github.com/pingcap/docs-cn/pull/1495/commits/8ea96d204e58ed7a18ca67875b63d679ffba950e

File: v3.0/reference/configuration/tidb-server/tidb-specific-variables.md (hunk 22 above)
Comment (ericsyh) [reply v3.0/reference/configuration/tidb-server/tidb-specific-variables.md]: Add back in https://github.com/pingcap/docs-cn/pull/1495/commits/8ea96d204e58ed7a18ca67875b63d679ffba950e

File: v3.0/reference/configuration/tidb-server/mysql-variables.md (hunk 23 above)
Comment (ericsyh) [reply v3.0/reference/configuration/tidb-server/mysql-variables.md]: Add back in https://github.com/pingcap/docs-cn/pull/1495/commits/8ea96d204e58ed7a18ca67875b63d679ffba950e

File: v3.0/reference/configuration/tidb-server/configuration.md (hunk 24 above)
Comment (ericsyh) [reply v3.0/reference/configuration/tidb-server/configuration.md]: Add back in https://github.com/pingcap/docs-cn/pull/1495/commits/8ea96d204e58ed7a18ca67875b63d679ffba950e

File: v3.0/reference/sql/character-set.md [hunk 25]
```diff
@@ -1,7 +1,7 @@
 ---
 title: 字符集支持
 category: reference
-aliases: ['/docs-cn/sql/character-set-support/']
+
```
Comment (dcalvin) [inline v3.0/reference/sql/character-set.md]:
```suggestion
aliases: ['/docs-cn/sql/character-set-support/']
```

File: v3.0/reference/configuration/tikv-server/configuration-file.md [hunk 26]
```diff
@@ -0,0 +1,1046 @@
+---
+title: TiKV 配置文件描述
+category: reference
+
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
```
Remove this blank line. And do it for other similar places.

File: v3.0/reference/configuration/tikv-server/configuration.md [hunk 27]
```diff
@@ -1,7 +1,7 @@
 ---
 title: TiKV 配置参数
 category: reference
-aliases: ['/docs-cn/op-guide/tikv-configuration/']
+aliases: ['/docs/op-guide/tikv-configuration/']
```
Comment (dcalvin) [inline v3.0/reference/configuration/tikv-server/configuration.md]:
```suggestion
aliases: ['/docs-cn/op-guide/tikv-configuration/']
```

**Comment by dcalvin** [comment]
Need to resolve conflicts before merge. LGTM now.

File: dev/TOC.md [hunk 28]
```diff
@@ -212,6 +212,7 @@
     - [生成列](reference/sql/generated-columns.md)
     - [分区表](reference/sql/partitioning.md)
     - [字符集](reference/sql/character-set.md)
+    - [SQL模式](reference/sql/sql-mode.md)
```
Comment (lilin90) [inline dev/TOC.md]: SQL模式 -> SQL 模式

File: dev/how-to/maintain/identify-slow-queries.md [hunk 29]
```diff
@@ -253,3 +253,21 @@ admin show slow top all 5
 ```
 
 由于内存限制，保留的慢查询记录的条数是有限的。当命令查询的 `N` 大于记录条数时，返回的结果记录条数会小于 `N`。
+
+输出内容详细说明，如下:
```
Comment (lilin90) [inline dev/how-to/maintain/identify-slow-queries.md]: : -> ：

File: dev/reference/configuration/pd-server/configuration.md [hunk 30]
```diff
@@ -102,3 +102,8 @@ PD 可以通过命令行参数或环境变量配置。
 + 指定 PD 使用的 namespace 分类器。
 + 默认："table"
 + 如果 TiKV 不与 TiDB 集群配合运行，建议配置为 'default'。
+
+## `--metrics-addr`
+
++ prometheus pushgateway 的地址，默认不向 promethus 推送数据.
```
Comment (lilin90) [inline dev/reference/configuration/pd-server/configuration.md]: - -> Prometheus Pushgateway
- -> Prometheus

File: dev/reference/sql/sql-mode.md [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+* ANSI: 符合标准 SQL ，对数据进行校验，如果不符合定义类型或长度，对数据类型调整或截断保存，且返回warning警告。
+* STRICT_TRANS_TABLES:  严格模式，对数据进严格校验，但数据出现错误时，插入到表中，并且返回错误。
+* TRADITIONAL: 采用此模式使 TiDB 的行为象 "传统" SQL 数据库系统，当在列中插入不正确的值时“给出错误而不是警告”，一旦发现错误立即放弃INSERT/UPDATE。
+
+## SQL mode 列表，如下：
+
+| 名称 | 含义 |
+| --- | --- |
+| PIPES_AS_CONCAT | 将`||`视为字符串连接操作符（＋）(同CONCAT())，而不视为OR（支持) |
```
Comment (lilin90) [inline dev/reference/sql/sql-mode.md]: This causes a display error at the website.

File: v3.0/TOC.md [hunk 32]
```diff
@@ -212,6 +212,7 @@
     - [生成列](reference/sql/generated-columns.md)
     - [分区表](reference/sql/partitioning.md)
     - [字符集](reference/sql/character-set.md)
+    - [SQL模式](reference/sql/sql-mode.md)
```
Comment (lilin90) [inline v3.0/TOC.md]: ditto

**Pull request merged**